bpf/process/types/basic.h

```diff
@@ -631,9 +632,101 @@ copy_char_buf(void *ctx, long off, unsigned long arg, int argm,
 	return __copy_char_buf(ctx, off, arg, bytes, has_max_data(argm), e, data_heap);
 }
 
+static inline __attribute__((always_inline)) long
+filter_char_buf_equal(struct selector_arg_filter *filter, char *arg_str, uint orig_len)
+{
+	__u32 *map_ids = (__u32 *)&filter->value;
+	void *string_map;
+	__u32 map_idx;
+	__u8 len;
+	__u8 padded_len;
+	int index;
+	char *heap, *zero_heap;
+	int zero = 0;
+
+	if (orig_len >= STRING_MAPS_SIZE_6 || !orig_len)
+		return 0;
+
+	len = (__u8)orig_len;
+	// Calculate padded string length
+	padded_len = len;
+	if (len % STRING_MAPS_KEY_INC_SIZE != 0)
+		padded_len = ((len / STRING_MAPS_KEY_INC_SIZE) + 1) * STRING_MAPS_KEY_INC_SIZE;
+
+	// Check if we have entries for this padded length.
+	// Do this before we copy data for efficiency.
+	index = (padded_len / STRING_MAPS_KEY_INC_SIZE) - 1;
+	map_idx = map_ids[index & 0x7];
```

Comment:
What is the map_idx here? Is this how we distinguish between different matchArgs? It's not entirely clear to me how we do that.

Reply:
I added a comment per @olsajiri's comment to make this more clear.